File: torrefactora/src/main/java/com/torrefactora/entity/Task.java
package com.torrefactora.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "task")
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_task")
    private Integer id;

    private String name;

    private String description;

    private LocalDateTime beginDate;

    private LocalDateTime endDate;

    @ManyToOne
    @JoinColumn(name = "id_priority_fk", referencedColumnName = "id_priority")
    private Priority priority;

    @ManyToOne
    @JoinColumn(name = "id_status_fk", referencedColumnName = "id_status")
    private Status status;
}

File: torrefactoraUi/src/app/model/task.ts
import { Priority } from 'src/app/model/priority';
import { Status } from 'src/app/model/status';

export class Task {

  id : number;
  name: string;
  description: string;
  beginDate: Date;
  endDate: Date;
  priority: Priority;
  status: Status

}
